### PR TITLE
Make auto-rendered product page image clickable

### LINKED ISSUE #231: Make auto-generated product page image clickable
When clicking on product image on auto-generated product page, the user should be directed to the product admin view

### PR DESCRIPTION
**Pull request by PeeterPaal**
Closes #231

### CODE REVIEW

File: sources/javascripts/concat/base.js [hunk 1]
```diff
@@ -569,6 +569,18 @@
     }
   };
 
+  // ===========================================================================
+  // Opens product admin view on product image click
+  // ===========================================================================
+
+  var handleProductImageClick = function(url) {
```
Comment (tanelj) [inline sources/javascripts/concat/base.js:576]: Let's rename as it actually is: `id` or slug not `url`.

```suggestion
  var handleProductImageClick = function(product_id) {
```

File: sources/javascripts/concat/base.js [hunk 2]
```diff
@@ -569,6 +569,18 @@
...
+    if (editmode()) {
+      $('.product-content .item-top').click(function() {
```
Comment (tanelj) [inline sources/javascripts/concat/base.js:578]: `item-top` is quit random target. Let's add `product-image` class to product image div.

File: layouts/auto_generated_product.tpl [hunk 3]
```diff
@@ -119,7 +119,8 @@
 
     <script>
       if (template) {
-       template.handleProductPageContent();
+        template.handleProductPageContent();
+        template.handleProductImageClick("{{ product.id }}");
```
Comment (tanelj) [inline layouts/auto_generated_product.tpl:123]: Let's add this line only in edit mode. No need to present in live html body. Also no need to add quotes arround id (that is number).
```suggestion

        {%- if product and editmode %}
          template.handleProductImageClick({{ product.id }});
        {%- endif %}
```

**Review by tanelj — CHANGES_REQUESTED** [review]
Works as expected., but some naming comments.

**Review by tanelj — APPROVED** [review]
Nice!

**Review by pxska — APPROVED** [review]

**Pull request merged**